app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -491,17 +483,73 @@ else if (possiblePoke[1] < 2) {
     /**
      * @return the likely pokemon number against the char sequence as well as the similarity
      */
-    private int[] getPossiblePokemon(CharSequence rhs) {
+    private int[] getPossiblePokemon(CharSequence poketext, CharSequence candytext) {
         int pokeNumber = 0;
         int bestMatch = 100;
-        for (int i = 0; i < pokeCalculator.pokedex.size(); i++) {
-            int similarity = pokeCalculator.get(i).getDistance(rhs);
-            if (similarity < bestMatch) {
-                pokeNumber = i;
-                bestMatch = similarity;
+
+        for (Pokemon p: pokeCalculator.pokedex) {
+            int dist = p.getDistance(poketext);
+            if (dist < bestMatch) {
+                pokeNumber = p.number;
+                bestMatch = dist;
+            }
+
+            /* We found a perfect match. We are done. */
+            if (bestMatch == 0) {
+                int[] result = {pokeNumber, bestMatch};
+                return result;
+            }
+        }
```

Comment:
The zero check is the performance optimization that says that if you already found a perfect match, the candy search can't do any better.

Also, the optimization isn't hypothetical. The distance check is very expensive and in a mobile app, you need to consider these things for the sake of power too.